gemstone/generator/generator.py

```diff
@@ -6,27 +6,49 @@
 from .port_reference import *
 
 
+__DEBUG_MODE = True
+
+
+def set_debug_mode(value: bool = True):
+    global __DEBUG_MODE
+    assert value in {True, False}
+    __DEBUG_MODE = value
+
+
+def get_debug_mode():
+    return __DEBUG_MODE
+
+
 class Generator(ABC):
+    __cache = {}
+
     def __init__(self, name=None):
         """
         name: Set this parameter to override default name for instance
         """
         self.ports = DotDict()
         self.wires = []
         self.instance_name = name
+        # use class name as a hash init
+        self.__hash = hash(self.__class__.__name__)
 
     @abstractmethod
     def name(self):
         pass
 
     def add_port(self, name, T):
-        if name in self.ports:
+        if get_debug_mode() and name in self.ports:
```

Comment:
why only do this in debug mode?

Reply:
Just to avoid search. I guess it's cheap enough to have it.